parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputCommitter.java

```
@@ -49,29 +50,48 @@ public void commitJob(JobContext jobContext) throws IOException {
+
+    if (level != JobSummaryLevel.NONE) {
       try {
         final FileSystem fileSystem = outputPath.getFileSystem(configuration);
         FileStatus outputStatus = fileSystem.getFileStatus(outputPath);
-        List<Footer> footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus);
+        List<Footer> footers;
+
+        if (level == JobSummaryLevel.ALL) {
+          footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus, false); // don't skip row groups
+        } else if (level == JobSummaryLevel.COMMON_ONLY) {
+          footers = ParquetFileReader.readAllFootersInParallel(configuration, outputStatus, true); // skip row groups
+        } else {
+          throw new IllegalArgumentException("Unrecognized job summary level: " + level);
+        }
+
         // If there are no footers, _metadata file cannot be written since there is no way to determine schema!
         // Onus of writing any summary files lies with the caller in this case.
         if (footers.isEmpty()) {
           return;
         }
+
         try {
-          ParquetFileWriter.writeMetadataFile(configuration, outputPath, footers);
+          ParquetFileWriter.writeMetadataFile(configuration, outputPath, footers, level);
         } catch (Exception e) {
-          LOG.warn("could not write summary file for " + outputPath, e);
+          LOG.warn("could not write summary file(s) for " + outputPath, e);
+
           final Path metadataPath = new Path(outputPath, ParquetFileWriter.PARQUET_METADATA_FILE);
           if (fileSystem.exists(metadataPath)) {
             fileSystem.delete(metadataPath, true);
           }
+
+          final Path commonMetadataPath = new Path(outputPath, ParquetFileWriter.PARQUET_COMMON_METADATA_FILE);
+          if (fileSystem.exists(commonMetadataPath)) {
+            fileSystem.delete(commonMetadataPath, true);
+          }
         }
       } catch (Exception e) {
```

same here, it would be better to let the exception propagate instead of doing a catch-all